### ISSUE TITLE
False positive warning when re-exporting custom derive without using `#[macro_use]`

### ISSUE THREAD

**Issue by xfix** [rust-lang/rust#43497]
The following code warns in a crate, with `example_derive` being a custom derive crate.

```rust
extern crate example_derive;

pub use example_derive::*;
```

The retrieved warning is:

```
warning: proc macro crates and `#[no_link]` crates have no effect without `#[macro_use]`
```

But that's not true, as `derive` coming from proc macro crate was re-exported.

This issue can be seen in https://github.com/xfix/enum-map when those 2 lines are removed: https://github.com/xfix/enum-map/blob/088f635d0cc40dabc7f4f1bd1ed27128e81aff93/enum-map/src/lib.rs#L43-L44 (workaround for this issue).

**Comment by DiThi** [comment]
I'm having the same issue with procedural macros, and the workaround doesn't seem to work. It also warns me of the unused import in the `use` line.

**Comment by xfix** [comment]
I created a crate as a workaround for this issue: https://crates.io/crates/reexport-proc-macro

**Comment by xfix** [comment]
Appears to be fixed, maybe with Edition 2018?
